utils/zerovec/src/varzerovec/databake.rs

```diff
@@ -2,42 +2,85 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+use crate::vecs::{Index16, Index32};
 use crate::{ule::VarULE, VarZeroSlice, VarZeroVec};
 use databake::*;
 
-impl<T: VarULE + ?Sized> Bake for VarZeroVec<'_, T> {
+impl<T: VarULE + ?Sized> Bake for VarZeroVec<'_, T, Index16> {
     fn bake(&self, env: &CrateEnv) -> TokenStream {
         env.insert("zerovec");
         if self.is_empty() {
-            quote! { zerovec::VarZeroVec::new() }
+            quote! { zerovec::vecs::VarZeroVec16::new() }
         } else {
             let bytes = databake::Bake::bake(&self.as_bytes(), env);
             // Safe because self.as_bytes is a safe input
-            quote! { unsafe { zerovec::VarZeroVec::from_bytes_unchecked(#bytes) } }
+            quote! { unsafe { zerovec::vecs::VarZeroVec16::from_bytes_unchecked(#bytes) } }
```

Comment:
nit: we should just use VZV here, the generic gets defaulted

Reply:
thought: we could implement for all formats and then do a typeid type check to emit the compact thing for VZV16

Reply:
It doesn't, that's the problem. It gets inferred, so it's very unsafe to not specify it.

Reply:
@robertbastian hmm, I guess.